src/eval/mod.rs

```diff
@@ -317,6 +317,45 @@ where
         }
 
         clos = match &*shared_term {
+            Term::Sealed(_, inner, lbl) => {
+                let stack_item = stack.peek_op_cont();
+                match stack_item {
+                    Some(OperationCont::Op2Second(BinaryOp::Unseal(), _, _, _)) => {
+                        clos = Closure {
+                            body: RichTerm {
+                                term: shared_term,
+                                pos,
+                            },
+                            env,
+                        };
+                        update_thunks(&mut stack, &clos);
+                        continuate_operation(clos, &mut stack, &mut call_stack)?
+                    }
+                    Some(OperationCont::Op1(UnaryOp::Seq(), _)) => {
+                        let update_closure = Closure {
+                            body: RichTerm {
+                                term: shared_term.clone(),
+                                pos,
+                            },
+                            env: env.clone(),
+                        };
+                        // Update the original thunk (the thunk which holds the result of the `Seq`) immediately.
+                        // We do this because  we are on a `Sealed` term, and this is in WHNF, and if we don't,
+                        // we will be unwrapping a `Sealed` term and assigning the "unsealed" value to the result
+                        // of the `Seq` operation. See also: https://github.com/tweag/nickel/issues/123
+                        update_thunks(&mut stack, &update_closure);
```

Comment:
Nitpick: if we are updating thunks in both cases (or erroring out), we can do it before the match.

```suggestion
                let closure = Closure {
                    body: RichTerm {
                        term: shared_term,
                        pos,
                    },
                    env,
                 };
                 // Update the original thunk (the thunk which holds the result of the op) in both cases,
                 // even if we continue with a seq.
                 // We do this because  we are on a `Sealed` term, and this is in WHNF, and if we don't,
                 // we will be unwrapping a `Sealed` term and assigning the "unsealed" value to the result
                 // of the `Seq` operation. See also: https://github.com/tweag/nickel/issues/123
                update_thunks(&mut stack, &clos);
                match stack_item {
                    Some(OperationCont::Op2Second(BinaryOp::Unseal(), _, _, _)) => {
                        continuate_operation(closure, &mut stack, &mut call_stack)?
                    }
                    Some(OperationCont::Op1(UnaryOp::Seq(), _)) => {
```